training/distributed_training/pytorch/model_parallel/gpt2/data_pipeline.py

```diff
@@ -0,0 +1,170 @@
+import gzip
+import json
+import os
+import h5py
+from typing import List, Tuple
+import random
+
+import numpy as np
+import smdistributed.modelparallel.torch as smp
+import torch
+
+
+class WikiPretrainingDataset(torch.utils.data.Dataset):
+    def __init__(self, input_file, max_pred_length):
+        self.input_file = input_file
+        self.max_pred_length = max_pred_length
+        f = h5py.File(input_file, "r")
+        keys = [
+            "input_ids",
+            "input_mask",
+            "segment_ids",
+            "masked_lm_positions",
+            "masked_lm_ids",
+            "next_sentence_labels",
+        ]
+        self.inputs = [np.asarray(f[key][:]) for key in keys]
+        f.close()
+
+    def __len__(self):
+        "Denotes the total number of samples"
+        return len(self.inputs[0])
+
+    def __getitem__(self, index):
+        [
+            input_ids,
+            input_mask,
+            segment_ids,
+            masked_lm_positions,
+            masked_lm_ids,
+            next_sentence_labels,
+        ] = [
+            torch.from_numpy(input[index].astype(np.int64))
+            if indice < 5
+            else torch.from_numpy(np.asarray(input[index].astype(np.int64)))
+            for indice, input in enumerate(self.inputs)
+        ]
+
+        masked_lm_labels = torch.ones(input_ids.shape, dtype=torch.long) * -1
+        index = self.max_pred_length
+        # store number of  masked tokens in index
+        padded_mask_indices = (masked_lm_positions == 0).nonzero(as_tuple=False)
+        if len(padded_mask_indices) != 0:
+            index = padded_mask_indices[0].item()
+        masked_lm_labels[masked_lm_positions[:index]] = masked_lm_ids[:index]
+
+        return [input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels]
+
+
+###### Load Openwebtext pretraining data ######
+class OpenwebtextPretrainingDataset(torch.utils.data.Dataset):
```

Comment:
Please add a list in the PR description of all datasets being used in this example. Each one must be reviewed and approved.